azavea/cac-tripplanner
# Feature/kak/multiple event destinations#1107

## Overview

Allow an event to be associated with multiple destinations, in order to support the upcoming "tours" feature.


### Demo

The field for associated destinations in the `admin` site UI for editing an event has changed from a drop-down to a multi-select field:
![image](https://user-images.githubusercontent.com/960264/62326451-1d2ddc80-b47c-11e9-8ccd-5ab7333d2266.png)


### Notes

Originally I was planning on doing #1125 as well as part of this task, but after discussion with @lederer decided to break that out into a separate task, and also added #1126 and #1127. Instead, this PR only handles the data migration and minimal logic changes to prevent errors. There are a couple of places where an event is still assumed to have a single destination associated with it (if any), where for now it simply uses the first destination for the event: in routing, and in setting properties such as flags.

There is a backwards migration, but for it to work, the old `destination` field must be added back alongside the new `destinations` field in the model before running the backward migration.

I also fixed a small issue with two of the images for the default development event not being wide enough; they were set by the migration, but editing the event later would fail validation on the image.


## Testing Instructions

 * Run migrations (provisioning will also do this)
 * Go to http://localhost:8024/admin/destinations/event and go to default event
 * The previous destination (Independence Seaport Museum) should be set on the new field
 * Edit multi-select widget to add other destinations and save
 * Home page should still load as expected
 * Routing should still work for event (it should go to one of the selected destinations)
 * Details page for event should still load as expected
 * Search results at http://localhost:8024/api/destinations/search should be as expected


Closes #1107
